atomicfu-transformer/src/main/kotlin/kotlinx/atomicfu/transformer/AtomicFUTransformerJS.kt

```diff
@@ -26,8 +26,8 @@ private const val FILL = "fill"
 private const val GET_ELEMENT = "get\\\$atomicfu"
 
 private val MANGLE_VALUE_REGEX = Regex(".${Pattern.quote(MANGLED_VALUE_PROP)}")
-private val ARRAY_GET_ELEMENT_REGEX = Regex(".$GET_ELEMENT\\(([A-Za-z0-9$]+)\\)")
-private val ARRAY_GET_ELEMENT_VALUE_REGEX = Regex("$ARRAY_GET_ELEMENT_REGEX.${Pattern.quote(MANGLED_VALUE_PROP)}")
+// matches index until the first occurence of ')', parenthesised index expressions not supported
+private val ARRAY_GET_ELEMENT_REGEX = Regex(".$GET_ELEMENT\\(([^)]*)\\)")
```

Comment:
For now, only simple non-parenthesised expressions are supported as an index.
Can I make a recursive regular expression like this: 
`\((?>[^()]|(?R))*\)` that would match a single pair of parentheses with any symbols in between including balanced parentheses to support more complex expressions:
`arr[atomic1.value]`, `arr[f(i) + f(j)]`..?